### PR TITLE
MWPW-146963: OST to support Annual template

### PR DESCRIPTION
**Pull request by 3ch023**
This PR includes 2 changes to OST:
https://git.corp.adobe.com/wcms/tacocat.js/pull/611
https://git.corp.adobe.com/wcms/tacocat.js/pull/614
First is a new 'P3Y' plantype. Second is a new 'Annual' template for ABM prices.

### How to test
To test OST on the branch follow these instructions: https://git.corp.adobe.com/wcms/tacocat.js/wiki/Testing-OST-Milo

###  Resolves: 
https://jira.corp.adobe.com/browse/MWPW-146963 
https://jira.corp.adobe.com/browse/MWPW-147541

**Test URLs:**
- Before: https://main--milo--adobecom.hlx.page/drafts/mariia/pr/price-cta?martech=off
- After: https://ost-annual--milo--adobecom.hlx.page/drafts/mariia/pr/price-cta?martech=off

### CODE REVIEW

**Comment by 3ch023** [comment]
one known issue for P3Y is that plantype is not displayed on a product yet.
image

this is because https://www.stage.adobe.com/special/tacocat/products.js is not updated yet, the automatic job will regenerate it.
`"PA-989":{"name":"Adobe Acrobat Classic","arrangement_code":"PA-989","icon":"https://mcs.odin.adobe.com/content/dam/mcs/en/icons/raw/svg/acrobat_pro_appicon_256.svg","planTypes":{"undefined":true}`

File: libs/blocks/merch/merch.js [hunk 1]
```diff
@@ -415,24 +422,8 @@ export async function getPriceContext(el, params) {
...
-      break;
-    case PRICE_TEMPLATE_STRIKETHROUGH:
-    case 'priceStrikethrough':
-      template = PRICE_TEMPLATE_STRIKETHROUGH;
-      break;
-    default:
-      break;
-  }
+  // The PRICE_TEMPLATE_MAPPING supports legacy OST links
+  const template = PRICE_TEMPLATE_MAPPING.get(params.get('type')) ?? params.get('type');
```
Comment (npeltier) [inline libs/blocks/merch/merch.js:426]: this is not exactly the same. we were defaulting to PRICE_TEMPLATE_REGULAR before, and we are not anymore

**Review by npeltier — CHANGES_REQUESTED** [review]

File: libs/blocks/merch/merch.js (hunk 1 above)
Comment (3ch023) [reply libs/blocks/merch/merch.js:426]: yes, i found this behaviour quite unexpected actually.
are you sure its a good idea to default 'invalid' template to 'price'?
i can revert it back but i can't think of when would it be a good idea

Comment (npeltier) [reply libs/blocks/merch/merch.js:426]: i can't tell you for sure until it's an actual bug :) 
it's just that behaviour change makes me nervous

**Review by bozojovicic — APPROVED** [review]

**Review by yesil — APPROVED** [review]

**Review by npeltier — APPROVED** [review]
lgtm

**Review by mokimo — APPROVED** [review]

**Comment by 3ch023** [comment]
@afmicka , i've just updated PR to include this revert https://git.corp.adobe.com/wcms/tacocat.js/pull/619
for price labels. but could you please double-check on the test page that it works as expected?

**Review by narcis-radu — APPROVED** [review]

**Pull request merged**